### ISSUE TITLE
asset_get_name function

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Feature-Requests#1488]
### Is your feature request related to a problem?

I saw a comment on #1479 that mentioned GM not having a asset_get_name() function, and realized I don't think there's a request for that yet, so figured I'd file one. This function wouldn't have been possible before, but with the ref changes, it should be possible now. Combined with the existing asset_get_type, this would streamline things like save systems.

### Describe the solution you'd like

Add a new function, asset_get_name. This would function similar to existing functions like object_get_name, sprite_get_name, etc. but it would take any asset type. It would essentially be the true opposite to asset_get_index.

**Comment by BackYard321** [comment]
Definitely! With the addition of #1318 (`asset_get_ids()`), this would eliminate the need for functions like `object_get_name()`, `sprite_get_name()`, `script_get_name()`, etc, in many cases.

**Comment by WangleLine** [comment]
I agree, it would probably be nicer to deprecate all the specific asset type name fetchy functions and just have one that works for all asset types instead.

**Comment by Totobal5** [comment]
yes please depecrate all the other and just make this one the default

**Comment by FoxyOfJungle** [comment]
I agree with everyone here.
